daemons/controld/controld_join_client.c

```diff
@@ -146,6 +146,10 @@ void
 join_query_callback(xmlNode * msg, int call_id, int rc, xmlNode * output, void *user_data)
 {
     char *join_id = user_data;
+
+    /* @COMPAT Currently unused. Consider dropping, or keep for potential future
+     * use during rolling upgrades.
+     */
     xmlNode *generation = create_xml_node(NULL, PCMK__XE_GENERATION_TUPLE);
```

Comment:
I believe this is used by do_dc_join_filter_offer()

Reply:
Maybe... I see what you're talking about. I'm not sure if `PCMK__XE_GENERATION_TUPLE` is making its way to the top level of `join_ack->xml`. That is testable by adding some tracing, or by manually tracing through the message/request passing code.

If it is, then it would be good to be more explicit in `do_dc_join_filter_offer()` so this doesn't happen again.

Reply:
Yeah it looked too painful to be sure ;)